### PR TITLE
Clean up and some followup tests for config-based inference.

### PR DESCRIPTION
**Pull request by mingxingtan**
Mainly follow up https://github.com/google/automl/pull/309

### CODE REVIEW

**Comment by mingxingtan** [comment]
@Ely-S Please take a look this PR.
@fsx950223 This PR allows hparams to accept yaml file, where keys can be integer as well.

**Comment by fsx950223** [comment]
LGTM

**Comment by fsx950223** [comment]
Don't use JSON and YAML together

**Comment by mingxingtan** [comment]
Did not find a good way to dump to a string with yaml.

**Comment by mingxingtan** [comment]
NM, yaml support that, just need to change dumps to dump.

**Comment by mingxingtan** [comment]
Thanks for the quick review @fsx950223 

@Ely-S I am going to merge this PR, but feel free to add your comments, and we can update the code accordingly later.

File: efficientdet/main.py [hunk 1]
```diff
@@ -141,13 +141,11 @@ def main(argv):
...
-  if FLAGS.num_epochs:  # TODO: remove this flag after updating all docs.
+  if FLAGS.num_epochs:  # NOTE: remove this flag after updating all docs.
     config.num_epochs = FLAGS.num_epochs
 
-  if isinstance(config.image_size, str) and 'x' in config.image_size:
-    # image size is in format of width x height.
-    width, height = config.image_size.split('x')
-    config.image_size = (int(height), int(width))
+  # Parse image size in case it is in string format.
+  config.image_size = utils.parse_image_size(config.image_size)
```
Comment (Ely-S) [inline efficientdet/main.py:148]: nice refactor

File: efficientdet/g3doc/coco_ids.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+0: background
```
Comment (Ely-S) [inline efficientdet/g3doc/coco_ids.yaml:1]: This is much nicer than a JSON file since keys can be ints.

**Comment by Ely-S** [comment]
@mingxingtan This is great.

I see that `tf.io.GFile` is the preferred way to open files. I'll use that next time.

I'm going test this by running some training/inference typical at my day job.

File: efficientdet/hparams_config.py [hunk 3]
```diff
@@ -91,32 +94,45 @@ def override(self, config_dict_or_str):
...
+      if not attr.startswith('_'):
         config_dict[attr] = getattr(module, attr)
 
     return config_dict
 
-  def parse_from_str(self, config_str):
+  def parse_from_yaml(self, yaml_file_path: Text) -> Dict[Any, Any]:
+    """Parses a yaml file and returns a dictionary."""
+    with tf.io.gfile.GFile(yaml_file_path, 'r') as f:
+      config_dict = yaml.load(f, Loader=yaml.FullLoader)
```
Comment (Ely-S) [inline efficientdet/hparams_config.py:127]: If this library is pyyaml, `yaml.safe_load` is generally preferred.

**Comment by Ely-S** [comment]
@mingxingtan It seems like yaml is not part of the python standard library, so this change is breaking on my system.

```
  File "/home/eli/ondaka/edet/automl/efficientdet/hparams_config.py", line 28, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
```

How would you feel about adding a requirements.txt file?

File: efficientdet/hparams_config.py (hunk 3 above)
Comment (mingxingtan) [reply efficientdet/hparams_config.py:127]: You are right. safe_load seems to be better.

**Comment by mingxingtan** [comment]
@Ely-S Good idea, could you help preparer a requirements.txt file?

**Comment by Ely-S** [comment]
sure thing

**Pull request merged**